internal/controller/controllers/agent_controller.go

```diff
@@ -1086,18 +1086,6 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 	spec := agent.Spec
 	var err error
 
-	if internalHost.Approved != spec.Approved {
-		err = r.Installer.UpdateHostApprovedInternal(ctx, internalHost.InfraEnvID.String(), agent.Name, spec.Approved)
```

Comment:
I'm suggesting to return an finish the reconcile after this call. 
This way the reconcile will be short and no races will be done no?

Reply:
When should the host update be done ?

Reply:
The fix should verify that the approved flag is set not before all other host updates